src/MOL_discretization.jl

```diff
@@ -1,22 +1,236 @@
 struct MOLFiniteDifference{T} <: DiffEqBase.AbstractDiscretization
-  dxs::T
-  order::Int
+    dxs::T
+    order::Int
+    MOLFiniteDifference(args...;order=2) = new{typeof(args[1])}(args[1],order)
 end
-MOLFiniteDifference(args...;order=2) = MOLFiniteDifference(args,order)
 
+# Get boundary conditions from an array
+function get_bcs(bcs,tdomain,domain)
+    lhs_deriv_depvars_bcs = Dict()
+    no_bcs = size(bcs,1)
+    for i = 1:no_bcs
+        var = bcs[i].lhs.op
+        if var isa Variable
+            var = var.name
+            if !haskey(lhs_deriv_depvars_bcs,var)
+                lhs_deriv_depvars_bcs[var] = Array{Expr}(undef,3)
+            end
+            j = 0
+            if isequal(bcs[i].lhs.args[1],tdomain.lower) # u(t=0,x) 
+                j = 1    
+            elseif isequal(bcs[i].lhs.args[2],domain.lower) # u(t,x=x_init)
+                j = 2
+            elseif isequal(bcs[i].lhs.args[2],domain.upper) # u(t,x=x_final)
+                j = 3
+            end
+            if bcs[i].rhs isa ModelingToolkit.Constant
+                lhs_deriv_depvars_bcs[var][j] = :(var=$(bcs[i].rhs.value))
+            else    
+                lhs_deriv_depvars_bcs[var][j] = Expr(bcs[i].rhs)
+            end
+        end
+    end
+    return lhs_deriv_depvars_bcs
+end
+
+
+# Recursively traverses the input expression (rhs), replacing derivatives by
+# finite difference schemes. It returns a time dependent expression (expr)
+# that will be evaluated in the "f" ODE function (in DiffEqBase.discretize),
+# Note: 'non-derived' dependent variables are inserted into the diff. equations
+#       E.g. Dx(u(t,x))=v(t,x)*Dx(u(t,x)), v(t,x)=t*x
+#            =>  Dx(u(t,x))=t*x*Dx(u(t,x))
+                            
+function discretize_2(input,deriv_order,approx_order,dx,X,len,
+                      deriv_var,dep_var_idx,indep_var_idx)
+    if input isa ModelingToolkit.Constant
+        return :($(input.value)) 
+    elseif input isa Operation
+        if input.op isa Variable
+            expr = :(0.0)
+            var = input.op.name
+            if haskey(indep_var_idx,var) # ind. var.
+                if var != :(t)
+                    i = indep_var_idx[var]
+                    expr = :($X[$i][2:$len[$i]-1])
+                else
+                    expr = :(t)
+                end
+            else # dep. var.
+                # TODO: time and cross derivatives terms
+                i = indep_var_idx[deriv_var[1]]
+                j = dep_var_idx[var]
+                if deriv_order == 0
+                    expr = :(u[:,$j])
+                elseif deriv_order == 1
+                    # TODO: approx_order and forward/backward should be 
+                    #       input parameters of each derivative
+                    approx_order = 1
+                    L = UpwindDifference(deriv_order,approx_order,dx[i],len[i]-2,-1)
+                    expr = :(-1*($L*Q[$j]*u[:,$j]))
+                elseif deriv_order == 2

```

Comment:
The difference here isn't 1 or 2, but more generally, when it's odd you want to use upwinding and when it's even you want to use centered difference.